[DOCS-14082] Restructure Agentic Onboarding setup page (#36655)
DataDog/documentation: content/en/agentic_onboarding/setup.md

{{< code-block lang="text" >}}Add Datadog Error Tracking to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Real User Monitoring" %}}
{{< code-block lang="text" >}}Add Datadog Real User Monitoring to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Product Analytics" %}}
{{< code-block lang="text" >}}Add Datadog Product Analytics to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Kubernetes" %}}
{{< code-block lang="text" >}}Add Datadog for Kubernetes to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Docker" %}}
{{< code-block lang="text" >}}Add Datadog for Docker to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Serverless" %}}
{{< code-block lang="text" >}}Instrument my AWS Lambda functions with Datadog{{< /code-block >}}
{{% /tab %}}
{{< /tabs >}}

The agent detects your stack, requests permission before each tool call, applies changes locally (without committing them), and prints verification steps. After the agent completes, commit the changes to your repository and set any new environment variables (API keys, application IDs) in your production environment. Then see [Next steps](#next-steps) to confirm data is flowing.

## Next steps

> - Confirm data is flowing in the Datadog UI for the product you set up: [Error Tracking][6], [RUM > Applications][7], [Infrastructure > Hosts][8], [Serverless > Functions][9], or [Logs > Live Tail][10].

Author: @vidurkhannadatadog Tagging you for help identifying the correct UI pages for the other products supported. I can also just point to the landing pages.

- Instrument local infrastructure as code (Terraform, Helm, Kustomize, Ansible, Pulumi, raw Kubernetes manifests, Docker Compose files) by editing files in place
- Instrument local application code by adding SDK initialization and configuration for supported frontends and backends

### Install and run the CLI

1. Run the CLI with `npx`, passing `--site` to target your [Datadog site][16]. You have two options:

    **Option 1: Interactive setup.** If you don't have a Datadog account yet, or you want to choose your product interactively, run without a `--product` flag. The CLI steps you through account setup and product choice.

    ```shell
    npx @datadog/ai-setup-cli --site datadoghq.com
    ```

    Replace the value of `--site` with the [Datadog site][16] for your account: `datadoghq.com`, `us3.datadoghq.com`, `us5.datadoghq.com`, `datadoghq.eu`, `ap1.datadoghq.com`, or `ap2.datadoghq.com`.

    **Option 2: Direct setup.** If you already have a Datadog account and want to install a specific product, pass `--product` to skip product selection.

    ```shell
    npx @datadog/ai-setup-cli --site datadoghq.com --product <PRODUCT>
    ```

>     Replace `<PRODUCT>` with one of `infrastructure`, `apm`, `logs`, `rum`, `error-tracking`, `product-analytics`, `serverless`, `code-coverage`, `test-optimization`, or `llm-observability`.

Author: ```suggestion
    - Replace the value of `--site` with the [Datadog site](http://localhost:1313/getting_started/site/) for your account.
    - Replace `<PRODUCT>` with one of `infrastructure`, `apm`, `logs`, `rum`, `error-tracking`, `product-analytics`, `serverless`, `code-coverage`, `test-optimization`, or `llm-observability`.
```

{{% /tab %}}

{{% tab "Product Analytics" %}}
{{< code-block lang="text" >}}Add Datadog Product Analytics to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Kubernetes" %}}
{{< code-block lang="text" >}}Add Datadog for Kubernetes to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Docker" %}}
{{< code-block lang="text" >}}Add Datadog for Docker to my project{{< /code-block >}}
{{% /tab %}}

{{% tab "Serverless" %}}
{{< code-block lang="text" >}}Instrument my AWS Lambda functions with Datadog{{< /code-block >}}
{{% /tab %}}
{{< /tabs >}}

The agent detects your stack, requests permission before each tool call, applies changes locally (without committing them), and prints verification steps. After the agent completes, commit the changes to your repository and set any new environment variables (API keys, application IDs) in your production environment. Then see [Next steps](#next-steps) to confirm data is flowing.

## Next steps

- Confirm data is flowing in the Datadog UI for the product you set up: [Error Tracking][6], [RUM > Applications][7], [Infrastructure > Hosts][8], [Serverless > Functions][9], or [Logs > Live Tail][10].

> - For team-wide rollout, propagate environment variables through your secret manager.

Author: @vidurkhannadatadog This is pretty vague. Can we be more specific here? Some Qs:

* What does the CLI actually write to disk? A .env, a datadog-agent config, a k8s manifest, IaC files? "Propagate env vars" reads differently depending on the output.
* Is there a Datadog-recommended rollout path or doc we should link to instead of re-inventing here?
* What was the original intent? Was this about (a) sharing the API key across multiple devs, (b) deploying the instrumented app to production, or (c) running the CLI on teammates' machines?

Author: Looks like this got bumped when I pushed some commits. This is for the second bullet in the `## Next steps` section.

## AI Setup CLI

The Datadog AI Setup CLI is a standalone tool that runs in your terminal. Use it when you don't want to install an MCP server, or for tasks the MCP setup doesn't support, such as bootstrapping a Datadog account.

The CLI can:

- Create a Datadog account end-to-end from the terminal
- Link an existing Datadog account to your local environment
- Instrument local infrastructure as code (Terraform, Helm, Kustomize, Ansible, Pulumi, raw Kubernetes manifests, Docker Compose files) by editing files in place
- Instrument local application code by adding SDK initialization and configuration for supported frontends and backends

### Install and run the CLI

1. Run the CLI with `npx`, passing `--site` to target your [Datadog site][16]. You have two options:

    **Option 1: Interactive setup.** If you don't have a Datadog account yet, or you want to choose your product interactively, run without a `--product` flag. The CLI steps you through account setup and product choice.

    ```shell
    npx @datadog/ai-setup-cli --site datadoghq.com
    ```

>     Replace the value of `--site` with the [Datadog site][16] for your account: `datadoghq.com`, `us3.datadoghq.com`, `us5.datadoghq.com`, `datadoghq.eu`, `ap1.datadoghq.com`, or `ap2.datadoghq.com`.

Author: ```suggestion
    - Replace the value of `--site` with the [Datadog site][16] for your account: `datadoghq.com`, `us3.datadoghq.com`, `us5.datadoghq.com`, `datadoghq.eu`, `ap1.datadoghq.com`, or `ap2.datadoghq.com`.
```

>     - Replace the value of `--site` with the [Datadog site][16] for your account: `datadoghq.com`, `us3.datadoghq.com`, `us5.datadoghq.com`, `datadoghq.eu`, `ap1.datadoghq.com`, or `ap2.datadoghq.com`.

Author: ```suggestion
    Replace the value of `--site` with the [Datadog site][16] for your account: `datadoghq.com`, `us3.datadoghq.com`, `us5.datadoghq.com`, `datadoghq.eu`, `ap1.datadoghq.com`, or `ap2.datadoghq.com`.
```

1. Complete the OAuth flow in your browser when prompted. After authentication, point the CLI to your code repository. The CLI detects your project's frameworks, applies the required configuration, and provisions any necessary environment variables.

1. Commit the changes to your repository. You can edit the Datadog environment variables (API keys, application IDs) for your specific environment.

## MCP Server

The Datadog MCP Server exposes the `onboarding` toolset to any MCP-compatible coding assistant. After you install and authenticate the server, you instrument a project by typing a one-line prompt. The agent reads your code, calls MCP tools (with your permission), applies changes, and verifies the result.

The MCP Server supports the following frameworks:

| Product | Frameworks |
|---------|------------|
| Error Tracking, RUM, Product Analytics | Android, Angular, iOS, Next.js, React, Svelte, Vanilla JS, Vue |
| Infrastructure Monitoring (Kubernetes) | Terraform, Ansible, Kustomize, Helm, Pulumi, raw manifests |

> | Serverless Monitoring (AWS Lambda) | Terraform, AWS CDK, Serverless Framework, SAM |

Author: @vidurkhannadatadog Update to capture full product list.

Reviewer: added in a comment below https://github.com/DataDog/documentation/pull/36655#issuecomment-4549035154

---
title: Agentic Onboarding Setup
description: Instrument your applications with Datadog using the AI Setup CLI or the Datadog MCP Server.

---

## Overview

Agentic Onboarding is a set of AI-driven tools that automate Datadog instrumentation for your applications and infrastructure:

- [AI Setup CLI](#ai-setup-cli): Set up Datadog from a terminal, without an AI coding assistant.

> - [MCP Server](#mcp-server): Set up Datadog through an LLM coding assistant (such as Claude Code or Cursor), which handles framework detection and configuration from your IDE.

Reviewer 2: ```suggestion
- [MCP server](#mcp-server): Set up Datadog through an LLM coding assistant (such as Claude Code or Cursor), which handles framework detection and configuration from your IDE.
```

Reviewer 2: Suggesting to only capitalize `server` when specifically referring to the `Datadog MCP Server`. For example, line 18 also leaves this un-capitalized.

Author: Applied in 602b5e6.

The two paths are complementary and use the same Datadog account. You can install the Datadog MCP Server in your IDE and run the CLI in a terminal.

## AI Setup CLI

> The Datadog AI Setup CLI is a standalone terminal tool. Use it when you don't want to install an MCP server, or for tasks the MCP setup doesn't support, such as bootstrapping a Datadog account.

Reviewer 2: ```suggestion
The Datadog AI Setup CLI is a standalone terminal tool. Use it when you don't want to install an MCP server, or for tasks the MCP setup doesn't support, such as creating a Datadog account.
```

Author: Applied in 602b5e6.

The CLI can:

- Create a Datadog account end-to-end from the terminal
- Link an existing Datadog account to your local environment
- Instrument local infrastructure as code (Terraform, Helm, Kustomize, Ansible, Pulumi, raw Kubernetes manifests, Docker Compose files) by editing files in place
- Instrument local application code by adding SDK initialization and configuration for supported frontends and backends

### Prerequisites

- Node.js 20 or later
- A Datadog account (optional; the CLI can create one for you)

> 

Reviewer 2: ```suggestion

You don't need an existing Datadog account. The CLI can create one during setup.

```

Reviewer 2: It seems a bit confusing to have something that's optional listed in the prerequisites. Suggesting to drop the bullet in favor of this note, but it might also suffice to just drop the word `optional`.

Author: Applied — removed the bullet entirely in 602b5e6. The account-creation capability comes through in the intro and in step 1, so the note felt redundant.

### Supported products

The CLI can set up the following products:

| Product | Identifier |
|---------|------------|
| App and API Protection | `app_and_api_protection` |
| Code Coverage | `ci_code_coverage` |
| Docker | `docker` |
| Error Tracking | `error-tracking` |

> | Infrastructure Monitoring | `infra-monitoring` |

Reviewer 2: There's some inconsistency in the `Identifier` formatting — some use underscores, and others hyphens. Are these pending confirmation from product as mentioned in the description's `Additional notes`?

> Pending confirmation from the product team on the exact CLI --product flag values for the new products before flipping to ready-for-merge.

Author: The identifiers are confirmed by the product team — the inconsistency (underscores vs. hyphens) reflects the actual flag values.

    ```

    - Replace the value of `--site` with the [Datadog site][16] for your account.
    - Replace `<PRODUCT>` with one of the [supported products](#supported-products).

1. Press <kbd>Enter</kbd> at the welcome screen and choose whether you have a Datadog account. A browser opens for OAuth (or for account creation if you don't have an account yet). Complete the flow and grant access to your Datadog account.

1. If you ran the CLI without `--product`, select what to set up from the product menu. (Direct setup with `--product` skips this menu.)

   {{< img src="agentic_onboarding/product-selection.png" alt="CLI menu 'What would you like to set up?' grouped by Infrastructure and Backend monitoring, Frontend Monitoring, LLM-Based applications, and CI Testing" style="width:80%;" >}}

   The CLI detects your project's frameworks, applies the required configuration, and provisions any necessary environment variables. Progress is reported stage by stage.

   {{< img src="agentic_onboarding/setup-example.png" alt="CLI showing 'Instrumenting your app, Stage 1 of 3: Datadog RUM (Real User Monitoring)' with progress steps" style="width:80%;" >}}

>    When setup completes, the CLI lists the products it instrumented and links to the Datadog UI to verify telemetry.

Reviewer 2: ```suggestion
   When setup completes, the CLI lists the products it instrumented and links to the Datadog UI to verify incoming data.
```

Reviewer 2: The Word List mentions to avoid using "telemetry" alone when referring to telemetry data.

Author: Applied in 602b5e6.

>    {{< img src="agentic_onboarding/success.png" alt="CLI showing 'Setup complete!' with check marks next to RUM, Error Tracking, and Product Analytics" style="width:80%;" >}}

Reviewer 2: ```suggestion
   {{< img src="agentic_onboarding/success.png" alt="CLI showing 'Setup complete!' with check marks next to RUM, Error Tracking, and Product Analytics." style="width:80%;" >}}
```

Author: Applied in 602b5e6.

1. Commit the changes to your repository. You can edit the Datadog environment variables (API keys, application IDs) for your specific environment.

After the CLI completes, see [Next steps](#next-steps).

> ## MCP Server

Reviewer 2: ```suggestion
## MCP server
```

Author: Applied in 602b5e6.

| Docker Observability | `docker-compose` and sidecar (`docker run`) deployments; Terraform, Ansible, and other IaC (Pulumi, CloudFormation, Puppet, Chef) |
| Linux Observability | Terraform, Ansible, other IaC (Pulumi, CloudFormation, Puppet, Chef), and plain-shell install |
| Serverless Monitoring (AWS Lambda) | AWS SAM, AWS CDK, Serverless Framework, Terraform, `datadog-ci lambda instrument` |
| Serverless Monitoring (GCP Cloud Run and Cloud Run Functions) | Terraform, `gcloud run deploy`, Cloud Run YAML, Dockerfile, Gen 2 `gcloud functions deploy` |
| LLM Observability | OpenAI, Anthropic, LangChain, Vercel AI SDK (auto-detected from project dependencies) |
| OpenTelemetry | Node.js / server-side TS, Browser JS / React / Vite, Python (Django, Flask, FastAPI), Java, Go |
| App and API Protection | Python, Node.js, Java, Go, Ruby, .NET, PHP (auto-detected from dependency manifests) |
| Code Coverage, Test Optimization | Jest, Vitest, Mocha, Playwright, Cypress, pytest, unittest, JUnit, TestNG, RSpec, minitest, xUnit, NUnit, MSTest v2, `go test`, XCTest / Swift Testing |

> ### Step 1: Install the MCP Server

Reviewer 2: ```suggestion
### Step 1: Install the MCP server
```

Author: Applied in 602b5e6.

{{< tabs >}}
{{% tab "Claude Code" %}}
In an active Claude Code session, run:

   <pre><code>claude mcp add --transport http datadog-onboarding-{{< region-param key="dd_datacenter_lowercase" >}} "{{< region-param key="mcp_server_endpoint" >}}?toolsets=onboarding"</code></pre>
{{% /tab %}}

{{% tab "Cursor" %}}
**Option 1: Install deeplink (recommended)**

> Click the install deeplink for your [Datadog site](/getting_started/site/), then confirm {{< ui >}}Install{{< /ui >}} for the **datadog-onboarding-{{< region-param key="dd_datacenter_lowercase" >}}** server when Cursor opens.

Reviewer 2: Is it intentional for this link to be defined inline, rather than as a reference link?

Author: Applied — changed to reference link [16] in 602b5e6.

   <pre><code>{{< region-param key="cursor_mcp_install_deeplink" >}}</code></pre>

**Option 2: Manual configuration**

Add the server to `~/.cursor/mcp.json`:

<pre><code>{
  "mcpServers": {
    "datadog-onboarding-{{< region-param key="dd_datacenter_lowercase" >}}": {
      "url": "{{< region-param key="mcp_server_endpoint" >}}?toolsets=onboarding"
    }
  }
}</code></pre>
{{% /tab %}}

{{% tab "Other MCP clients" %}}

> Any MCP client that supports HTTP transport can connect to the Datadog MCP Server. Point it at the endpoint for your [Datadog site](/getting_started/site/):

Reviewer 2: Likewise confirming if this link is intentionally defined inline.

Author: Applied — changed to reference link [16] in 602b5e6.

   <pre><code>{{< region-param key="mcp_server_endpoint" >}}?toolsets=onboarding</code></pre>

{{% /tab %}}
{{< /tabs >}}

> ### Step 2: Authenticate the MCP Server

Reviewer 2: ```suggestion
### Step 2: Authenticate the MCP server
```

Author: Applied in 602b5e6.

1. After you install the MCP Server, your coding assistant prompts you to authenticate. Press <kbd>Enter</kbd> to open the Datadog OAuth screen in your browser.

> 1. After authentication completes, choose {{< ui >}}Open{{< /ui >}} to return to your IDE and grant the MCP Server access to your Datadog account.

Reviewer 2: ```suggestion
1. After you install the MCP server, your coding assistant prompts you to authenticate. Press <kbd>Enter</kbd> to open the Datadog OAuth screen in your browser.
1. After authentication completes, choose {{< ui >}}Open{{< /ui >}} to return to your IDE and grant the MCP server access to your Datadog account.
```

Author: Applied in 602b5e6.

{{< code-block lang="shell" >}}npx @datadog/ai-setup-cli --product serverless --serverless-compute-type=aws-lambda{{< /code-block >}}

**GCP Cloud Run containers**
{{< code-block lang="text" >}}Add Datadog for GCP Cloud Run containers to my project{{< /code-block >}}

{{< code-block lang="shell" >}}npx @datadog/ai-setup-cli --product serverless --serverless-compute-type=gcp-cloud-run{{< /code-block >}}

**GCP Cloud Run functions**
{{< code-block lang="text" >}}Add Datadog for GCP Cloud Run functions to my project{{< /code-block >}}

{{< code-block lang="shell" >}}npx @datadog/ai-setup-cli --product serverless --serverless-compute-type=gcp-cloud-run-functions{{< /code-block >}}

{{% /tab %}}

{{< /tabs >}}

The agent detects your stack, requests permission before each tool call, applies changes locally (without committing them), and prints verification steps.

After the agent completes, commit the changes to your repository and set any new environment variables (API keys, application IDs) in your production environment. Then see [Next steps](#next-steps) to confirm data is flowing.

## Next steps

> Confirm data is flowing in the Datadog UI for the product you set up: [Error Tracking][6], [RUM > Applications][7], [Infrastructure > Hosts][8], [Serverless > Functions][9], or [Logs > Live Tail][10].

Reviewer 2: ```suggestion
Confirm data is flowing in the Datadog UI for the product you set up:

  - [Error Tracking][6]
  - [RUM > Applications][7]
  - [Infrastructure > Hosts][8]
  - [Serverless > Functions][9]
  - [Logs > Live Tail][10]

```

Reviewer 2: Recommending to convert this to a list for easier visual parsing and to enhance AI retrieval ([preview](https://docs-staging.datadoghq.com/domalessi/docs-14082-agentic-onboarding/agentic_onboarding/setup/?tab=claudecode&site=us#next-steps)).

Author: Applied in 602b5e6.

---
title: Agentic Onboarding Setup
description: Instrument your applications with Datadog using the AI Setup CLI or the Datadog MCP Server.

---

{{< site-region region="gov,gov2" >}}
<div class="alert alert-danger">Agentic Onboarding is not available in the selected site ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

## Overview

Agentic Onboarding is a set of AI-driven tools that automate Datadog instrumentation for your applications and infrastructure:

- [AI Setup CLI](#ai-setup-cli): Set up Datadog from a terminal, without an AI coding assistant.

> - [MCP Server](#mcp-server): Set up Datadog through an LLM coding assistant (such as Claude Code or Cursor), which handles framework detection and configuration from your IDE.

Reviewer 2: Just flagging what looks like some inconsistent terminology for the coding assistant — `AI coding assistant` becomes `LLM coding assistant` in the next line. I'm also seeing `MCP-compatible coding assistant` (lines 97 and 101) and just `coding assistant` (line 159). 

For clarity and to assist with AI retrieval, it might be helpful to pick one and use it throughout — `coding assistant` might suffice, with `MCP-compatible` being added where MCP compatibility is the point.

Author: Applied — standardized to "coding assistant" throughout in 602b5e6.

    ```shell
    npx @datadog/ai-setup-cli --site datadoghq.com
    ```

    Replace the value of `--site` with the [Datadog site][16] for your account: `datadoghq.com`, `us3.datadoghq.com`, `us5.datadoghq.com`, `datadoghq.eu`, `ap1.datadoghq.com`, or `ap2.datadoghq.com`.

    **Option 2: Direct setup.** If you already have a Datadog account and want to install a specific product, pass `--product` to skip product selection.

    ```shell
    npx @datadog/ai-setup-cli --site datadoghq.com --product <PRODUCT>
    ```

    - Replace the value of `--site` with the [Datadog site][16] for your account.
    - Replace `<PRODUCT>` with one of the [supported products](#supported-products).

1. Press <kbd>Enter</kbd> at the welcome screen and choose whether you have a Datadog account. A browser opens for OAuth (or for account creation if you don't have an account yet). Complete the flow and grant access to your Datadog account.

1. If you ran the CLI without `--product`, select what to set up from the product menu. (Direct setup with `--product` skips this menu.)

>    {{< img src="agentic_onboarding/product-selection.png" alt="CLI menu 'What would you like to set up?' grouped by Infrastructure and Backend monitoring, Frontend Monitoring, LLM-Based applications, and CI Testing" style="width:80%;" >}}

Reviewer 2: ```suggestion
   {{< img src="agentic_onboarding/product-selection.png" alt="CLI menu 'What would you like to set up?' grouped by Infrastructure and Backend monitoring, Frontend Monitoring, LLM-Based applications, and CI Testing." style="width:80%;" >}}
```

   The CLI detects your project's frameworks, applies the required configuration, and provisions any necessary environment variables. Progress is reported stage by stage.

>    {{< img src="agentic_onboarding/setup-example.png" alt="CLI showing 'Instrumenting your app, Stage 1 of 3: Datadog RUM (Real User Monitoring)' with progress steps" style="width:80%;" >}}

Reviewer 2: ```suggestion
   {{< img src="agentic_onboarding/setup-example.png" alt="CLI showing 'Instrumenting your app, Stage 1 of 3: Datadog RUM (Real User Monitoring)' with progress steps." style="width:80%;" >}}
```

npx @datadog/ai-setup-cli --product serverless --serverless-compute-type=aws-lambda
```

**GCP Cloud Run containers**
{{< code-block lang="text" >}}Help me monitor my GCP Cloud Run services with Datadog{{< /code-block >}}

```shell
npx @datadog/ai-setup-cli --product serverless --serverless-compute-type=gcp-cloud-run
```

**GCP Cloud Run functions**
{{< code-block lang="text" >}}Help me monitor my GCP Cloud Run functions with Datadog{{< /code-block >}}

```shell
npx @datadog/ai-setup-cli --product serverless --serverless-compute-type=gcp-cloud-run-functions
```

{{% /tab %}}

{{< /tabs >}}

The agent detects your stack, requests permission before each tool call, applies changes locally (without committing them), and prints verification steps.

After the agent completes, commit the changes to your repository and set any new environment variables (API keys, application IDs) in your production environment. Then see [Next steps](#next-steps) to confirm data is flowing.

## Next steps

- Confirm data is flowing in the Datadog UI for the product you set up: [Error Tracking][6], [RUM > Applications][7], [Infrastructure > Hosts][8], [Serverless > Functions][9], or [Logs > Live Tail][10].

> - For team-wide rollout, propagate environment variables through your secret manager.

Author: ```suggestion
```

## Overview

Agentic Onboarding lets LLM coding agents instrument your applications and infrastructure for [Error Tracking][1], [Real User Monitoring (RUM)][2], [Product Analytics][3], [Infrastructure Monitoring][4], and [Serverless Monitoring][5].

> There are three ways to get started. Pick the one that matches how you work:

Author: Should this section call out prereqs for each path? For example: Node.js for the Setup CLI, an existing Datadog account + role permissions for the MCP Server, a Claude subscription (and which Claude surface) for Skills. Or fine to leave implicit / assume the audience will know what's required for each?

Reviewer: Pre requisites:
1. for CLI: Node.js version 20 and above
2. for MCP: no need as we map them to specific agents already like claude, cursor etc.

| Path | Use when |
|------|----------|
| [Setup CLI](#setup-cli) | You want a standalone command-line tool (no coding assistant required) for first-time account creation, linking an existing account, or instrumenting local infrastructure-as-code (IaC) and application code from a terminal. |
| [MCP Server](#mcp-server) | You work with an LLM coding assistant (such as Claude Code or Cursor) and want it to detect frameworks, write configuration, and provision tokens directly from your IDE. |
| [Skills](#skills) | You're using Claude and want a focused skill for a single onboarding step (such as installing the Agent or enabling a cloud integration) rather than a full MCP toolset. |

> The three paths are complementary and work against the same Datadog account. You can install the Datadog MCP Server in your IDE, run the CLI in a terminal, and invoke skills from Claude.

Author: Each section below has a different internal structure: Setup CLI opens with a "what it can do" bullet list, MCP Server opens with a Supported frameworks table, Skills opens with an Available skills table. Probably intentional since the paths genuinely differ, but flagging in case we want to harmonize the openers.

Reviewer: Let' frame each section like:
1. What it can do?
2.  Pre req (if any)
3. supported frameworks/products
4. How to set it up / use the tool
5. Next steps

<div class="alert alert-info">This section is in progress. Additional onboarding skills are planned, and the content below is subject to change before general availability.</div>

### Available skills

| Skill | What it does |
|-------|--------------|
| Install Datadog Agent | Detects the host or cluster type and installs the Datadog Agent with the right configuration. |
| Enable cloud integration | Wires up AWS, GCP, or Azure cloud integrations against your Datadog account. |

### Invoke a skill

Skills are discoverable inside Claude. From a Claude conversation:

1. Reference the skill by name (for example, "install the Datadog Agent on this host").
1. Claude loads the skill, asks for any required credentials or scope (such as host or cluster type, or cloud account ID), and walks through the steps.
1. The skill prints verification commands when it finishes.

Skills work standalone, so you don't need the MCP Server installed to use them. After the skill completes, see [Next steps](#next-steps).

> ## Next steps

Author: Your original draft had a placeholder for Troubleshooting in Next steps that I dropped. Can add back when we create a Troubleshooting page.
